src/main/kotlin/org/rust/ide/inspections/RustSuspiciousAssignmentInspection.kt

```diff
@@ -0,0 +1,79 @@
+package org.rust.ide.inspections
+
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiElementVisitor
+import org.rust.ide.inspections.fixes.SubstituteTextFix
+import org.rust.lang.core.psi.*
+
+/**
+ * Checks for use of the non-existent =*, =! and =- operators that are probably typos but can be compiled.
+ * Analogue of Clippy's suspicious_assignment_formatting.
+ * QuickFix 1: Change `a =? b` to `a ?= b`
+ * QuickFix 2: Change `a =? b` to `a = ?b`
+ */
+class RustSuspiciousAssignmentInspection : RustLocalInspectionTool() {
+    private object Constants {
+        val LONG_TEXT_THRESHOLD = 10
+        val LONG_TEXT_SUBST = ".."
+    }
+
+    override fun getDisplayName(): String = "Suspicious assignment"
+
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
+        return object : RustElementVisitor() {
+            override fun visitBinaryExpr(expr: RustBinaryExprElement) {
+                if (expr.operator.text == "=") {
```

Comment:
I'm totally ok with `if (!cond) return` in such cases, just usually try to follow a more conservative rule of having only one `return` per function in a new project until I get familiar with the adopted code style : )
I'll do the refactoring.